### PR TITLE
`podman events`: check for an error after we finish reading events

### PR DESCRIPTION
**Pull request by nalind**
#### Does this PR introduce a user-facing change?

```release-note
`podman events` should be able to diagnose invalid values being passed as arguments to its `--since` and `--until` flags.
```

### CODE REVIEW

File: cmd/podman/system/events.go [hunk 1]
```diff
@@ -162,15 +163,21 @@ func eventsCmd(cmd *cobra.Command, _ []string) error {
...
 		select {
 		case event, ok := <-eventChannel:
 			if !ok {
 				// channel was closed we can exit
+				select {
+				case err := <-errChannel:
+					if err != nil {
+						return err
+					}
+				case <-time.After(1 * time.Millisecond):
```
Comment (Luap99) [inline cmd/podman/system/events.go:179]: assuming you did this to get a non blocking call on the error channel you should use the `default` case instead

Comment (nalind) [reply cmd/podman/system/events.go]: Either one assumes that the error will be ready by the time the `select` statement looks at the channel, but yeah, `default` would be cleaner.  Changed it.

**Comment by mheon** [comment]
I hate the nested select, but I don't see another way.

LGTM once you get the test to pass.

File: pkg/bindings/system/system.go [hunk 2]
```diff
@@ -42,6 +43,27 @@ func Events(ctx context.Context, eventChan chan types.Event, cancelChan chan boo
...
+		}
+		if response.Status != "" {
+			if bodyText != "" {
+				return fmt.Errorf("%s: %s", response.Status, bodyText)
+			}
+			return errors.New(response.Status)
+		}
+		if bodyText != "" {
+			return fmt.Errorf("server responded with unexpected status code %d: %s", response.StatusCode, bodyText)
+		}
```
Comment (Luap99) [inline pkg/bindings/system/system.go:63]: the rest of the code simply calls `return response.Process(nil)` to convert the error

Comment (nalind) [reply pkg/bindings/system/system.go]: Oh, that's _much_ better.  Thanks!

**Review by Luap99 — APPROVED** [review]
/lgtm

**Pull request merged**